mParticle-Taplytics/MPKitTaplytics.m

```diff
@@ -198,39 +198,52 @@ - (nonnull MPKitExecStatus *)onLogoutComplete:(FilteredMParticleUser *)user requ
 }
 
 #pragma mark e-Commerce
-- (MPKitExecStatus *)logCommerceEvent:(MPCommerceEvent *)commerceEvent {
-    MPKitExecStatus *execStatus = [[MPKitExecStatus alloc] initWithSDKCode:[[self class] kitCode] returnCode:MPKitReturnCodeSuccess forwardCount:0];
-    
+
+- (MPKitExecStatus *)routeCommerceEvent:(MPCommerceEvent *)commerceEvent {
+    MPKitExecStatus *execStatus = [self createStatus:MPKitReturnCodeSuccess];
+
     if (commerceEvent.action == MPCommerceEventActionPurchase) {
         MPTransactionAttributes *transaction = commerceEvent.transactionAttributes;
         if (transaction != nil && transaction.revenue != nil && transaction.transactionId != nil) {
             [Taplytics logRevenue:transaction.transactionId revenue:transaction.revenue];
             [execStatus incrementForwardCount];
         }
-        
     } else {
         NSArray *expandedInstructions = [commerceEvent expandedInstructions];
-        
+
         for (MPCommerceEventInstruction *commerceEventInstruction in expandedInstructions) {
-            [self logEvent:commerceEventInstruction.event];
+            [self routeEvent:commerceEventInstruction.event];
             [execStatus incrementForwardCount];
         }
     }
-    
+
     return execStatus;
 }
 
 #pragma mark Events
-- (MPKitExecStatus *)logEvent:(MPEvent *)event {
+- (MPKitExecStatus *)routeEvent:(MPEvent *)event {
     NSString * eventName = event.name;
-    NSDictionary * metaData = event.info;
+    NSDictionary * metaData = event.customAttributes;
     if (metaData != nil) {
         [Taplytics logEvent:eventName value:nil metaData:metaData];
+    } else if (eventName != nil) {
```

Comment:
Will line 228 crash if eventName is nil? Maybe it should be
```
    if (eventName != nil) {
            if (metaData != nil) {
                [Taplytics logEvent:eventName value:nil metaData:metaData];
            } else {
                [Taplytics logEvent:eventName];
            }
    } else {
        return [self createStatus:MPKitReturnCodeFail];
    }
```

Reply:
it shouldn't crash if eventName is nil, but I'll change it anyway for brevity